### PR TITLE
chore: refactor to expose fcm at usecase level

### PR DESCRIPTION
**Pull request by maxwellgithinji**
# Review Checklist

[Bugs](https://shields.io/)
[Features](https://shields.io/)
[Tests](https://shields.io/)

#### Summary*

- [ ] fix: Bug fix for `Add the items tackled here as checklists`
- [x] feat: Completed task
- [ ] chore: Incomplete task
  - [ ] test: Sub-task 1

#### Structure*

- [ ] The Pull Request has a `proper` title that conforms to our [MR title standards](https://gist.github.com/mikepea/863f63d6e37281e329f8)
- [ ] The Pull Request has one commit, and if there are more than one, they should be squashed
- [ ] The commit should have a proper title and a short description
- [ ] The commit must be signed off
- [ ] Unused imports are not present
- [ ] Dead code is not present
- [ ] Ensure dry running library to confirm changes

#### Tests

- [ ] Proper and high quality unit, integration and acceptance(if applicable) tests have been written
- [ ] The coverage threshold should not be lowered

### Sign off*

- [ ] All comments have been resolved by the reviewers
- [ ] Approved by Czar {replace_with_czar_name}
- [ ] Signed off by second reviewer {replace_with_name}
- [ ] Ensure all checks are done before merging :warning:
- [ ] All PRs needs to be signed before merging :warning:

#### N/B:

- Add a checklist if more than one item is done.
- Add screenshots and/or images where necessary
- Indicate any breakages caused in the UI :exclamation:
- Where necessary, indicate which issue the Pull Request solves (Closes #)
- Any new files are updated in the folder structure in the [README](../../README.md)

### CODE REVIEW

File: pkg/engagement/usecases/usecases.go [hunk 1]
```diff
@@ -2,23 +2,30 @@ package usecases
...
+	"github.com/savannahghi/engagement/pkg/engagement/usecases/fcm"
 	"github.com/savannahghi/engagement/pkg/engagement/usecases/feed"
 )
 
 // Interactor is an implementation of the usecases interface
 type Interactor struct {
 	*feed.UseCaseImpl
 	*feed.NotificationImpl
+	*fcm.PushImpl
+	*fcm.FCMImpl
```
Comment (Kenneth-Mathenge) [inline pkg/engagement/usecases/usecases.go]: Let's just have one `fcm` usecase.

File: pkg/engagement/usecases/usecases.go [hunk 2]
```diff
@@ -2,23 +2,30 @@ package usecases
...
 }
 
 // NewInteractor initializes a new usecases interactor
 func NewInteractor(infrastructure infrastructure.Interactor) Interactor {
```
Comment (Kenneth-Mathenge) [inline pkg/engagement/usecases/usecases.go]: This was a bit of of an oversight from my end. Rename this to `NewUsecasesInteractor`

File: pkg/engagement/usecases/fcm/push.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import (
+	"context"
+
+	"github.com/savannahghi/engagement/pkg/engagement/infrastructure"
+	"github.com/savannahghi/firebasetools"
+)
+
+// PushUsecase defines FCM push usecases interface
+type PushUsecase interface {
+	PushFn(
```
Comment (Kenneth-Mathenge) [inline pkg/engagement/usecases/fcm/push.go]: Renaming things. `push` - `pushFn` might cause some interesting problems. Avoid it if it's not necessary.

File: README.md [hunk 4]
```diff
@@ -36,6 +36,21 @@ A cleanly architected project should be:
...
+
+## How to deploy changes for this projects
+
+This is a library that uses semantic versioning hence every change made to the library, after approval and release, it must be tagged, then upgrade the (engagement-service)[https://github.com/savannahghi/engagement-service] with the latest tag version.
+
+tagging 
+```sh
+  git tag -m "v0.0.n" "v0.0.n"
+  git push --tags
+ ```
```
Comment (Kenneth-Mathenge) [inline README.md:53]: For now this README will pass. But it needs an MR of it's own. Focus on the moving. Then come back to this.

File: pkg/engagement/usecases/usecases.go [hunk 5]
```diff
@@ -2,23 +2,30 @@ package usecases
...
 	notification := feed.NewNotification(infrastructure)
 	feed := feed.NewFeed(infrastructure)
+	push := fcm.NewPush(infrastructure)
+	fcm := fcm.NewFCM(infrastructure)
 
 	return Interactor{
 		feed,
 		notification,
+		push,
+		fcm,
```
Comment (Kenneth-Mathenge) [inline pkg/engagement/usecases/usecases.go:26]: Moving this work systematically like this also affords you the luxury of fixing "minor" things we over look like `good doc strings` and `good tests (unit, integration and acceptance)`.

For every service you move make sure it adheres to these.

Comment (maxwellgithinji) [reply pkg/engagement/usecases/usecases.go:26]: for sure. noted

**Review by Kenneth-Mathenge — DISMISSED** [review]

**Review by Kenneth-Mathenge — APPROVED** [review]

**Pull request merged**